### ISSUE TITLE
@ mentions are marked as error

### ISSUE THREAD

**Issue by tiff** [languagetool-org/languagetool#1155]
If I mention someone in a text (e.g. on Slack) it is shown to me as an error. Seems to happen with all languages.

![bildschirmfoto 2018-08-23 um 21 16 19](https://user-images.githubusercontent.com/37363/44546953-ccfdb480-a719-11e8-8ee0-3c4cf7a45402.png)

**Comment by MikeUnwalla** [comment]
Frequently, the word that comes after the @ is not a dictionary word. Thus, if we always check the word, we will get many false positives (the current situation). If we always ignore the word after @, there are 2 problems:
- We do not find typographic errors. If I accidentally write '@cat' when I mean 'cat', LT will not find the error.
- We do not give a possibly useful message for misspelled dictionary words that are in @ mentions.

My preference: the default spelling should always ignore @text. Add a rule which the user can select/deselect. The rule has these mutually exclusive user-selectable options:
- Always give a message for a word that starts with @ (=the current situation).
- If the text after @ gives no spelling error, do nothing (=check the term and ignore the @).
- If the text after @ gives a spelling error, give a message to say, "make sure this part of the @ mention is correct".

**Comment by Discostu36** [comment]
> We do not find typographic errors. If I accidentally write '@cat' when I mean 'cat', LT will not find the error.

I don't think this is very likely. As you need to use the AltGr/CMD key for entering an `@`, I don't think I've ever entered it accidentally.

>  We do not give a possibly useful message for misspelled dictionary words that are in @ mentions.

But how often are there dictionary words in `@` mentions? As we can see in this group, dictionary words as nicknames are not very common.

So in my opinion, all terms starting with an `@` should be ignored.

**Comment by MikeUnwalla** [comment]
> I don't think this is very likely.

Agreed. My example is bad. I mean that LT should not assume that all @text is correct. As a professional editor, I want to be sure that I can find all possible errors in a text. If LT makes assumptions, I cannot be sure.

If a document is scanned using optical character recognition, I cannot be sure that the text is correct. I can easily envisage the software changing 'Chat' to '@hat' or 'draft' to '@raft'.

(Aside: my comment about user-selectable rules applies also to e-mail addresses and websites.)
